lib/fathead/sass/parse_functions.py

```diff
@@ -0,0 +1,264 @@
+
+    def parse_for_data(self):
+        """
+        Main gateway into parsing the data. Will retrieve all necessary data elements.
+        """
+        data = []
+        
+        for function_section in self.function_sections:
+            function = self.parse_for_function_name(function_section[0])
+            if function:
+                method_signature = self.parse_for_method_signature(function_section[0])
+                description = self.parse_for_description(function_section[1])
+                anchor = self.parse_for_anchor(function_section[0])
+
+                url = self.create_url(anchor)
+
+                data_elements = {
+                    'function': function,
+                    'method_signature': method_signature,
+                    'description': description,
+                    'url': url
+                }
+
+                data.append(data_elements)
+
+        self.parsed_data = data
+
+    def get_data(self):
+        """
+        Get the parsed data.
+        Returns:
+            self.parsed_data: Dict containing necessary data elements
+        """
+        return self.parsed_data
+
+
+class DataOutput(object):
+    """
+    Object responsible for outputting data into the output.txt file
+    """
+    def __init__(self, data):
+        self.data = data
+
+    def create_names_from_data(self, data_element):
+        """
+        Figure out the name of the function. Will contain the module name if one exists.
+        Args:
+            data_element: Incoming data dict
+
+        Returns:
+            Name, with whitespace stripped out
+
+        """
+        function = data_element.get('function')
+
+        dotted_name = '{}{}{}'.format(function, '.' if function  else '', function)
+        spaced_name = '{} {}'.format(function, function)
+
+        return dotted_name.strip(), spaced_name.strip()
+
+    def create_file(self):
+        """
+        Iterate through the data and create the needed output.txt file, appending to file as necessary.
+
+        """
+        with open('output.txt', 'a') as output_file:
+            for data_element in self.data:
+                if data_element.get('function'):
+                    method_signature = data_element.get('method_signature').encode('utf-8')
+                    description = data_element.get('description').encode('utf-8')
+                    name = data_element.get('function').encode('utf-8')
+                    abstract = str(method_signature) + '<br>' + str(description)
+                    url = data_element.get('url').encode('utf-8')
+                    list_of_data = [
+                        name,                       # unique name
+                        'A',                        # type is article
+                        '',                         # no redirect data
+                        '',                         # ignore
+                        '',                         # no categories
```

Comment:
Is there any way we can determine a category for each (or most) articles? For example if we know it's a function, like `abs()` or `rgb()` we should specify the category.

The result is that a query like "sass functions" will show a nice tile view that shows all the articles we've specified as "functions".

You could go one step further and specify the category for `rgba()` as `function, rgb` and then "sass rgb functions" could list out the specific RGB functions such as `rgb`, `rgba`, `red`, `green`, etc.

Does that make sense?